src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/EnvelopeRepositoryTest.java

```diff
@@ -200,6 +200,31 @@ void should_find_envelopes_by_container_and_file_name() {
         });
     }
 
+    @Test
+    void should_find_last_envelope_by_container_and_file_name() {
+        // given
+        final String fileName = "foo.bar";
+        final String container = "bar";
+
+        // and
+        repo.insert(new NewEnvelope(container, fileName, now(), now(), Status.DISPATCHED));
+        // this envelope should be selected
+        UUID envelopeId2 =  repo.insert(
```

Comment:
why 2?

Reply:
The second envelope, there should also be 1 and 3 but they are not used. If you do mind I will rename